### ISSUE TITLE
Vertical alignment of chips and input incorrect

### ISSUE THREAD

**Issue by CrocoDillon** [TeamWertarbyte/material-ui-chip-input#285]
Chips and input should be center aligned vertically but they are not. The offset is only small with normal chips but with small chips it's very obvious.

Sandbox: https://codesandbox.io/s/material-ui-chip-input-input-alignment-uvftp

Enter some data in the input field to see what I mean.

**Comment by leMaik** [comment]
Awesome! Added complexity due to small chips. :roll_eyes: 

We now have three input variants (normal, outlined, filled), two chip variants (small, normal) and three margins (normal, dense, none). That's 28 variants to check. I'll look into this soon and make a storybook example with all 28 cases.

**Comment by CrocoDillon** [comment]
Yeah I figured when doing some testing on Code Sandbox (never used it before btw, it's awesome) that small chips are new in Material UI v4.

Also... it's 2019 and still vertical centering can be a challenge 😆 Though flexbox does ease the pain a little bit in most cases.

**Comment by leMaik** [comment]
> Also... it's 2019 and still vertical alignment can be a challenge

Yeah, it's mostly a chip margin and line height thing here.

**Comment by hermanho** [comment]
How about applying css?
https://codesandbox.io/s/material-ui-chip-input-input-alignment-w8cej
